### PR TITLE
Add iSeq retrieval

### PR DESCRIPTION
**Pull request by ingkebil**
This PR solves the retrieval of the iSeq samplesheet.

**How to test?**
1. `bash update-demux-stage.sh isheet` on clinical-preproc stage.
1. run: `demux sheet fetch -a iseq --pad --indexlength 10 $flowcell`
        with $flowcell BPC29611-2526 or BPC29611-3324 (see comment below)

**Expected outcome**
The fecth command will print the samplesheet to stdout. The samplesheet should contain an index and index2 column. index should end with AT and index2 should start with AC.

Headers should be: 
Sample_Name | Sample_Plate | Description | I7_Index_ID | index | I5_Index_ID | index2 | Manifest | GenomeFolder | Sample_Project | Sample_Well

This is **minor version** bump: extending and not modifying public API.

@emiliaol can you provide a FC id?

- [x] Code reviewed by @ingkebil  
- [x] Tested by @patrikgrenfeldt and @emiliaol 
- [x] Deploy signature @ingkebil

### CODE REVIEW

**Comment by emiliaol** [comment]
BPA73008-1225
BPA73008-1825
BPC29611-2526
BPC29611-3324

There you go. 4 of them! But I think only the last two are useful as we may not have created the others in LIMS

**Comment by ingkebil** [comment]
Thanks!

**Comment by patrikgrenfeldt** [comment]
@ingkebil is it possible to create a test for this? What about having a test for iSeq in tests/utils/test_samplesheet.py?

**Comment by ingkebil** [comment]
Hi Patrik, yeas please add a unit test for this, wkr, kenny

**Comment by ingkebil** [comment]
In case it wasn't clear, I handed over this work to you on Wednesday ;)

**Comment by patrikgrenfeldt** [comment]
Configuration:
image
Command:
image
Output:
image
Expected outcome?
- [x] The fecth command will print the samplesheet to stdout. 
- [x] The samplesheet should contain an index and index2 column. 
- 👎  index should end with AT and index2 should start with AC.

**Comment by patrikgrenfeldt** [comment]
Command:
image
Output:
image
Expected outcome?
- [x] The fecth command will print the samplesheet to stdout. 
- [x] The samplesheet should contain an index and index2 column. 
- [x]  index should end with AT and index2 should start with AC.

**Comment by patrikgrenfeldt** [comment]
image

Headers should be:
- [x] Sample_ID | Sample_Name | Sample_Plate | Description | I7_Index_ID | index | I5_Index_ID | index2 | Manifest | GenomeFolder | Sample_Project | Sample_Well
data should be like
- [x] ACC5726A1 | ACC5726A1 |   | ACC5726A1 |   | TATCGCACAT |   | ACCTTAGTGT |   |   | 881352 |  👍

File: demux/cli/samplesheet.py [hunk 1]
```diff
@@ -219,6 +222,56 @@ def get_project(project):
...
+                     'sample_id',
+                     'i7_index_id',
+                     'index',
+                     'i5_index_id',
+                     'index2',
+                     'manifest',
+                     'genome_folder',
+                     'sample_name',
+                     'sample_well']
+        header = IseqSamplesheet.header_map.values()
```
Comment (ingkebil) [inline demux/cli/samplesheet.py]: Please define the header directly here

File: demux/cli/samplesheet.py [hunk 2]
```diff
@@ -227,4 +280,4 @@ def get_project(project):
         line['sample_name'] = project
 
         # print it!
-        click.echo(delimiter.join([str(line[head]) for head in lims_keys]))
+        click.echo(delimiter.join([str(line.get(lims_key, '')) for lims_key in lims_keys]))
```
Comment (ingkebil) [inline demux/cli/samplesheet.py]: please revert

Comment (ingkebil) [reply demux/cli/samplesheet.py]: to fail early

Comment (ingkebil) [reply demux/cli/samplesheet.py]: thanks

File: demux/utils/__init__.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
 # -*- coding: utf-8 -*-
 
-from .samplesheet import Samplesheet, HiSeqXSamplesheet, NIPTSamplesheet, HiSeq2500Samplesheet, MiseqSamplesheet, SampleSheetValidationException
+from .samplesheet import Samplesheet, HiSeqXSamplesheet, NIPTSamplesheet, HiSeq2500Samplesheet, \
+    MiseqSamplesheet, SampleSheetValidationException, IseqSamplesheet
```
Comment (ingkebil) [inline demux/utils/__init__.py]: remove iseqsamplesheet

File: demux/utils/samplesheet.py [hunk 4]
```diff
@@ -625,3 +625,20 @@ def to_demux(self, delim=',', end='\n'):
             rs.append(delim.join(line))
         return end.join(rs)
 
+
+class IseqSamplesheet(Samplesheet):
```
Comment (ingkebil) [inline demux/utils/samplesheet.py]: remove the whole class

File: demux/cli/samplesheet.py [hunk 5]
```diff
@@ -1,14 +1,16 @@
...
 import copy
 import csv
 import os
 
+import click
+
 from cglims.api import ClinicalLims, ClinicalSample
-from ..utils import Samplesheet, HiSeqXSamplesheet, NIPTSamplesheet, HiSeq2500Samplesheet, MiseqSamplesheet
+from ..utils import Samplesheet, HiSeqXSamplesheet, NIPTSamplesheet, HiSeq2500Samplesheet,\
+                    MiseqSamplesheet, IseqSamplesheet
```
Comment (ingkebil) [inline demux/cli/samplesheet.py]: remove iseqsamplesheet

Comment (patrikgrenfeldt) [reply demux/cli/samplesheet.py]: done

File: demux/cli/samplesheet.py (hunk 1 above)
Comment (patrikgrenfeldt) [reply demux/cli/samplesheet.py]: defined

File: demux/cli/samplesheet.py (hunk 2 above)
Comment (patrikgrenfeldt) [reply demux/cli/samplesheet.py]: reverted

File: demux/utils/__init__.py (hunk 3 above)
Comment (patrikgrenfeldt) [reply demux/utils/__init__.py]: removed

File: demux/utils/samplesheet.py (hunk 4 above)
Comment (patrikgrenfeldt) [reply demux/utils/samplesheet.py]: removed

**Comment by patrikgrenfeldt** [comment]
@ingkebil this has been tested OK on the iSeq machine, ready for merge and deploy?

**Comment by ingkebil** [comment]
Awesome! Merge and deploy :moneybag:

**Comment by ingkebil** [comment]
I'll merge and bump. Not sure where this needs to be deployed?

**Comment by emiliaol** [comment]
It would be nice to have on both hasta and thalamus

**Pull request merged**